[mono] iOS Simulator builds will pick up a system ICU in some circumstances

Error:
```
  ld: building for iOS Simulator, but linking in dylib built for macOS, file '/usr/local/opt/icu4c/lib/libicui18n.dylib' for architecture x86_64
  clang: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.
```

Env vars set (homebrew-recommended):
```
# icu4c is keg-only
export PATH="/usr/local/opt/icu4c/bin:$PATH"
export PATH="/usr/local/opt/icu4c/sbin:$PATH"
export LDFLAGS="-L/usr/local/opt/icu4c/lib $LDFLAGS"
export CPPFLAGS="-I/usr/local/opt/icu4c/include $CPPFLAGS"
export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig:$PKG_CONFIG_PATH"
```

It should probably just never check the system, given I would not expect there to ever be a system install of iOS ICU builds.

--- a/src/mono/CMakeLists.txt
+++ b/src/mono/CMakeLists.txt
@@ -652,14 +652,21 @@ endif()
 ######################################
 # ICU CHECKS
 ######################################
+if(HOST_OSX OR HOST_MACCAT OR HOST_IOS OR HOST_TVOS)
+  # FIXME: Handle errors
+  execute_process(COMMAND  brew --prefix OUTPUT_VARIABLE brew_prefix OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  if((HOST_MACCAT OR HOST_IOS OR HOST_TVOS) AND "${CMAKE_SHARED_LINKER_FLAGS}" MATCHES "${brew_prefix}/opt/icu4c/lib")
+    message(FATAL_ERROR "Linker flags contain the Homebrew version of ICU which conflicts with the iOS/tvOS/MacCatalyst version: ${CMAKE_SHARED_LINKER_FLAGS}")
+  endif()
+endif()
+
 set(ICU_SHIM_PATH "${CLR_SRC_NATIVE_DIR}/libs/System.Globalization.Native")
 if(MONO_CROSS_COMPILE)
 elseif(HOST_OSX AND NOT HOST_MACCAT)
   include(FindPkgConfig)
-  # FIXME: Handle errors
-  # Defines ICU_INCLUDEDIR/ICU_LIBDIR
-  execute_process(COMMAND  brew --prefix OUTPUT_VARIABLE brew_prefix OUTPUT_STRIP_TRAILING_WHITESPACE)
   set(ENV{PKG_CONFIG_PATH} "{$PKG_CONFIG_PATH}:${brew_prefix}/lib/pkgconfig:${brew_prefix}/opt/icu4c/lib/pkgconfig")
+  # Defines ICU_INCLUDEDIR/ICU_LIBDIR
   pkg_check_modules(ICU icu-uc)
   set(OSX_ICU_LIBRARY_PATH /usr/lib/libicucore.dylib)
   set(ICU_FLAGS "-DTARGET_UNIX -DU_DISABLE_RENAMING -Wno-reserved-id-macro -Wno-documentation -Wno-documentation-unknown-command -Wno-switch-enum -Wno-covered-switch-default -Wno-extra-semi-stmt -Wno-unknown-warning-option -Wno-deprecated-declarations")

